Done.
Setting up the random seed for reproducibility...
Done.
dynamic_import - class_name:  OracleTrainer
Initializing model...experiments third
Done.
Initializing datasets...
Done.
Initializing clients...
dynamic_import - class_name:  OracleClient
Done.
Initializing server...
dynamic_import - class_name:  OracleServer
Running without server optimizer
Done.
Initialize return score, metrics, ckpt, ckpt step...
Setting up metrics...
Done.
Done.
Initializing optimizer and scheduler...
Done.
dynamic_import - class_name:  OracleClient
The experiment begins...
ROUND 1/6: Training 2 Clients...
CLIENT 1/2: erfurt5 MIX
Clipping input data to the valid range for imshow with RGB data ([0..1] for floats or [0..255] for integers).
Clipping input data to the valid range for imshow with RGB data ([0..1] for floats or [0..255] for integers).
Clipping input data to the valid range for imshow with RGB data ([0..1] for floats or [0..255] for integers).
Clipping input data to the valid range for imshow with RGB data ([0..1] for floats or [0..255] for integers).
Clipping input data to the valid range for imshow with RGB data ([0..1] for floats or [0..255] for integers).
Clipping input data to the valid range for imshow with RGB data ([0..1] for floats or [0..255] for integers).
Clipping input data to the valid range for imshow with RGB data ([0..1] for floats or [0..255] for integers).
Clipping input data to the valid range for imshow with RGB data ([0..1] for floats or [0..255] for integers).
Exception ignored in: <function _MultiProcessingDataLoaderIter.__del__ at 0x7fdcb6227af0>
Traceback (most recent call last):
  File "/home/utente/anaconda3/envs/LADD/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 1328, in __del__
    self._shutdown_workers()
  File "/home/utente/anaconda3/envs/LADD/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 1288, in _shutdown_workers
    self._workers_done_event.set()
  File "/home/utente/anaconda3/envs/LADD/lib/python3.9/multiprocessing/synchronize.py", line 338, in set
    self._cond.notify_all()
  File "/home/utente/anaconda3/envs/LADD/lib/python3.9/multiprocessing/synchronize.py", line 297, in notify_all
    self.notify(n=sys.maxsize)
  File "/home/utente/anaconda3/envs/LADD/lib/python3.9/multiprocessing/synchronize.py", line 278, in notify
    while self._woken_count.acquire(False):
KeyboardInterrupt:
Traceback (most recent call last):
  File "/home/utente/Scrivania/PROVA/LADD/src/run.py", line 53, in <module>
    run_experiment()
  File "/home/utente/Scrivania/PROVA/LADD/src/run.py", line 21, in run_experiment
    max_score = trainer.train(*trainer.train_args, **trainer.train_kwargs)
  File "/home/utente/Scrivania/PROVA/LADD/src/federated/trainers/oracle_trainer.py", line 150, in train
    return self.perform_fed_oracle_training(
  File "/home/utente/Scrivania/PROVA/LADD/src/federated/trainers/oracle_trainer.py", line 104, in perform_fed_oracle_training
    losses = self.server.train_clients(partial_metric=partial_train_metric)
  File "/home/utente/Scrivania/PROVA/LADD/src/federated/servers/oracle_server.py", line 112, in train_clients
    out = c.train(partial_metric, r=r)
  File "/home/utente/Scrivania/PROVA/LADD/src/clients/oracle_client.py", line 228, in train
  File "/home/utente/Scrivania/PROVA/LADD/src/clients/oracle_client.py", line 198, in run_epoch
    profiler=profiler, stop_at_step=stop_at_step, r=r)
  File "/home/utente/Scrivania/PROVA/LADD/src/clients/oracle_client.py", line 161, in __exec_epoch
    plt.show()
KeyboardInterrupt